sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/restriction/ThroughputEstimatorTest.java

```diff
@@ -43,70 +45,71 @@ public void setup() {
 
   @Test
   public void testThroughputCalculation() {
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(2, 0), 10);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(3, 0), 20);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(5, 0), 30);
+    estimator.update(Timestamp.ofTimeSecondsAndNanos(10, 0), 40); // Exclusive
+    assertEquals(6D, estimator.getFrom(Timestamp.ofTimeSecondsAndNanos(11, 0)), DELTA);
```

Comment:
shouldn't this be (10+20+30+40)/10=10?

Reply:
Not with the current implementation, as a new second is not directly added. It is only considered when we move to the next second. So, 40 would only be included if we added something at the 11th second.

TBH, I am not sure why we are not immediately considering it, as this logic gets confusing. I will try and modify it.

Reply:
Modified it to include every byte reported instead of only considering the last ones when the window slides.